General/score.py

```diff
@@ -0,0 +1,96 @@
+""""
+this module defines the score functions we are using
+"""
+
+""""""""""""""""""""""""""""""""""""""""""" Imports """""""""""""""""""""""""""""""""""""""""""
+import abc
+import numpy as np
+import sklearn
+import pandas as pd
+from General.utils import TrainSamples
+import scipy.stats as stats
+
+""""""""""""""""""""""""""""""""""""""""""" Class """""""""""""""""""""""""""""""""""""""""""""
+
+
+class ScoreFunction:
+    """
+    An abstract class for ScoreFunction.
+    """
+
+    def __init__(self, learning_algorithm: sklearn.base.ClassifierMixin, alpha: int):
+        """"
+        :param learning_algorithm : a base algorithm in which the function rely on
+        :param alpha : represent the ratio between function's params
+        """
+        self._learning_algorithm = learning_algorithm
+        self._alpha = alpha
+
+    @abc.abstractmethod
+    def _execute_function(self):
+        """"
+        this function will execute the score function
+        """
+        ...
+
+    def __call__(self, *args, **kwargs):
+        return self._execute_function()
+
+
+class ScoreFunctionA(ScoreFunction):
+    """"
+    return the feature score according to the theory we explain in the PDF.
+    """
+    def __init__(self, train_samples: TrainSamples, given_features: list[int], new_feature: int,
```

Comment:
If there is an option to pass these parameters in _execute_function, I think this will be better since during the init phase od the algoritms (when we will initialize the score function class)- we won't have all this params. We will have it only in predict phase, when we will call _execute_function :)